config/initializers/rack_attack.rb

```diff
@@ -27,6 +27,13 @@ class Rack::Attack
     req.ip if req.path =~ /\A#{Regexp.escape('/api/v1/api_key')}/ && req.get?
   end
 
+  # Throttle PATCH and DELETE profile requests
+  throttle("clearance/remember_token", limit: 100, period: 10.minutes) do |req|
+    if req.path == "/profile" && (req.patch? || req.delete?)
+      req.cookies["remember_token"].presence
```

Comment:
This logic would be best implemented with `remember_token` as it ensures that requests are throttled for the specific compromised session. However, I am not sure if it is safe to store `remember_token` in our cache store. Alternatively, we can use `req.ip` here.